waf_test.go

```diff
@@ -67,6 +67,109 @@ func TestListWAFPackages(t *testing.T) {
 	assert.Error(t, err)
 }
 
+func TestWAFPackage(t *testing.T) {
+	setup()
+	defer teardown()
+
+	testZoneID := "abcd123"
+
+	handler := func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, "GET", r.Method, "Expected method 'GET', got %s", r.Method)
+
+		w.Header().Set("content-type", "application/json")
+		// JSON data from: https://api.cloudflare.com/#waf-rule-packages-properties
+		fmt.Fprintf(w, `{
+			"success": true,
+			"errors": [],
+			"messages": [],
+			"result":
+			{
+				"id": "a25a9a7e9c00afc1fb2e0245519d725b",
+				"name": "WordPress rules",
+				"description": "Common WordPress exploit protections",
+				"detection_mode": "traditional",
+				"zone_id": "023e105f4ecef8ad9ca31a8372d0c353",
+				"status": "active"
+			}
+		}`)
+	}
+
+	mux.HandleFunc("/zones/"+testZoneID+"/firewall/waf/packages/a25a9a7e9c00afc1fb2e0245519d725b", handler)
+
+	want := WAFPackage{
+		ID:            "a25a9a7e9c00afc1fb2e0245519d725b",
+		Name:          "WordPress rules",
+		Description:   "Common WordPress exploit protections",
+		ZoneID:        "023e105f4ecef8ad9ca31a8372d0c353",
+		DetectionMode: "traditional",
+		Sensitivity:   "",
+		ActionMode:    "",
+	}
+
+	d, err := client.WAFPackage(testZoneID, "a25a9a7e9c00afc1fb2e0245519d725b")
+
+	if assert.NoError(t, err) {
+		assert.Equal(t, want, d)
+	}
+
+	_, err = client.WAFPackage(testZoneID, "123")
```

Comment:
I'm not generally a fan of multiple assertions in a single test. Instead, explicit and isolated tests are my preferred option to ensure we can use specific test names and don't end up with a single test with lots of responsibilities. Not a hill I'm willing to die on here though :)

Reply:
seemed to be a common pattern in other similar tests -- there's probably several things we could do to clean up the WAF-related (and other) tests, but probably not in scope for this PR

Reply:
eg. https://github.com/cloudflare/cloudflare-go/pull/323/files#diff-caef407880318468a47a25739545707eR66